diesel/src/connection/statement_cache.rs

```diff
@@ -1,12 +1,70 @@
 use std::any::TypeId;
 use std::borrow::Cow;
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::hash::Hash;
 
 use backend::Backend;
 use query_builder::*;
 use result::QueryResult;
 
 #[doc(hidden)]
 #[allow(missing_debug_implementations)]
+pub struct StatementCache<DB: Backend, Statement> {
+    pub cache: RefCell<HashMap<StatementCacheKey<DB>, Statement>>,
+}
+
+impl<DB, Statement> StatementCache<DB, Statement> where
+    DB: Backend,
+    DB::TypeMetadata: Clone,
+    DB::QueryBuilder: Default,
+    StatementCacheKey<DB>: Hash + Eq,
+{
+    pub fn new() -> Self {
+        StatementCache {
+            cache: RefCell::new(HashMap::new())
+        }
+    }
+
+    pub fn len(&self) -> usize {
+        self.cache.borrow().len()
+    }
```

Comment:
I hadn't as you refactored bin/test for mysql at the same time. Gimme a sec to add it